### PR TITLE
Fix broken lints, update to pkg:lints, Bump SDK constraint

### PR DESCRIPTION
**Pull request by kevmoo**
Also cleaned out changelog

### CODE REVIEW

File: lib/src/file.dart [hunk 1]
```diff
@@ -398,9 +398,8 @@ class _FileSpan extends SourceSpanMixin implements FileSpan {
         sourceUrl == other.sourceUrl;
   }
 
-  // Eliminates dart2js warning about overriding `==`, but not `hashCode`
   @override
-  int get hashCode => super.hashCode;
+  int get hashCode => Object.hash(_start, _end, sourceUrl);
```
Comment (jakemac53) [inline lib/src/file.dart:402]: This looks like it was intentionally using the identity hash code? I would probably do some digging at least to understand why it was done the way it was.

Comment (kevmoo) [reply lib/src/file.dart:402]: It's not identity – it's the hash from the parent.

I wanted to avoid the warning.

I can put it back

Comment (jakemac53) [reply lib/src/file.dart:402]: I just assumed it was falling back on `Object.hashCode` so its identity - if its actually hitting some real super implementation that would be even more likely to be an intentional thing I would think - maybe the identity isn't intended to take these other fields into account etc.

Comment (jakemac53) [reply lib/src/file.dart:402]: Just to be clear in general we should probably have a real hash code implementation here, but somebody very intentionally didn't do that so I wouldn't want to change it without understanding why.

Comment (sigmundch) [reply lib/src/file.dart:402]: We may need a time machine and ask @kevmoo (see https://github.com/dart-lang/source_span/commit/3cf8669673b639b164e5f46fd358983051d15b0c)

Comment (jakemac53) [reply lib/src/file.dart:402]: Hah!

Comment (jakemac53) [reply lib/src/file.dart:402]: Actually given that I think its probably safe to just add the real implementation @kevmoo .

Comment (natebosch) [reply lib/src/file.dart:402]: Yeah the real implementation looks like the best choice.

**Review by natebosch — APPROVED** [review]

**Pull request merged**